src/Core/Service/ServiceSourceResolver.php

```diff
@@ -71,53 +65,32 @@ public function filesystem(Manifest|AppEntity $app): Filesystem
 
         /** @var ServiceSourceConfig $sourceConfig */
         $sourceConfig = $app->getSourceConfig();
+        $appInfo = AppInfo::fromNameAndSourceConfig($name, $sourceConfig);
 
-        return new Filesystem($this->checkVersionAndDownloadAppZip($name, $sourceConfig));
+        return $this->filesystemForVersion($appInfo);
     }
 
     public function reset(array $filesystems): void
     {
     }
 
-    /**
-     * @param ServiceSourceConfig $sourceConfig
-     */
-    private function checkVersionAndDownloadAppZip(string $serviceName, array $sourceConfig): string
-    {
-        try {
-            $client = $this->serviceClientFactory->fromName($serviceName);
-        } catch (ServiceException) {
-            // the service is not available, so we cannot download it.
-            // this can happen if the service is not installed or the service client is misconfigured. or the service has been removed.
-            return Path::join($this->temporaryDirectoryFactory->path(), $serviceName);
-        }
-
-        $latestAppInfo = $client->latestAppInfo();
-
-        if (!$this->isLatestVersionInstalled($latestAppInfo, $sourceConfig)) {
-            // the app revision has changed in the service, so we must update the app
-            // this can happen if the system attempts to download the app, before a service update rollout has completed
-            $this->eventDispatcher->dispatch(new ServiceOutdatedEvent($serviceName, Context::createDefaultContext()));
```

Comment:
Since this is removed, we should remove the event and subscriber: `\Shopware\Core\Service\Subscriber\ServiceOutdatedSubscriber`